File: problems/problems_LCR_079/solution.py
import solution
from typing import *


class Solution(solution.Solution):
    def solve(self, test_input=None):
        return self.subsets(test_input)

    def subsets(self, nums: List[int]) -> List[List[int]]:
        ans = []
        def backtrack(idx: int, path: List[int]):
            if idx == len(nums):
                ans.append(list(path))
                return
            # 不选当前数
            backtrack(idx + 1, path)
            # 选当前数
            path.append(nums[idx])
            backtrack(idx + 1, path)
            # backtrack
            path.pop()

        backtrack(0, [])
        return ans

File: problems/problems_LCR_079/solution.ts
function subsets(nums: number[]): number[][] {
	const ans: number[][] = [];
	const backtrack: Function = (idx: number, path: number[]): void => {
		if (idx === nums.length) {
			ans.push([...path]);
			return;
		}
		backtrack(idx + 1, path);
		path.push(nums[idx]);
		backtrack(idx + 1, path);
		path.pop();
	}
	backtrack(0, []);
	return ans;
};

export function Solve(inputJsonElement: string): any {
	const inputValues: string[] = inputJsonElement.split("\n");
	const nums: number[] = JSON.parse(inputValues[0]);
	return subsets(nums);
}
